contracts/src/v0.8/ccip/offRamp/EVM2EVMOffRamp.sol

```
@@ -343,6 +343,14 @@ contract EVM2EVMOffRamp is IAny2EVMOffRamp, AggregateRateLimiter, ITypeAndVersio
       (Internal.MessageExecutionState newState, bytes memory returnData) = _trialExecute(message, offchainTokenData);
       _setExecutionState(message.sequenceNumber, newState);
 
+      // Since it's hard to estimate whether manual execution will succeed, we
+      // revert the entire transaction if it fails. This will show the user if
+      // their manual exec will fail before they submit it.
+      if (manualExecution && newState == Internal.MessageExecutionState.FAILURE) {
+        // If manual execution fails, we revert the entire transaction.
+        revert ExecutionError(returnData);
+      }
```

an alternative is to only revert after every message has been executed and one of them does not succeed, e.g. add the following to manual exec
```
for (uint256 i = 0; i < numMsgs; ++i) {
      Internal.MessageExecutionState state = getExecutionState(report.messages[i].sequenceNumber);
      if (state != Internal.MessageExecutionState.SUCCESS) revert ExecutionError("");
}
```
if only attempting single message (assume most common with manual exec), it is better to fail early with error data so I agree with the approach in this PR, wanna put it here to discuss in case we missed anything

---

Yeah considered it but this is cleaner and solves 99% of the cases